### PR TITLE
all: Use new pacemaker option to not manage stateless a/a services

### PR DESCRIPTION
**Pull request by vuntz**
When this option is used, only the following resources stay managed by
pacemaker (assuming the relevant roles are added to the cluster):

 - rabbitmq
 - database
 - haproxy
 - keystone-fernet-rotate
 - neutron-ha-tool
 - ceilometer-central
 - cinder-volume

All other services basically work in the same way as without HA, with a
simple systemd service.

This reduces the pacemaker complexity. This removes self-healing (can be
added back with Restart= in systemd services), and the ordering
constraints that are cluster-wide, but limits the impact of pacemaker,
and side-effects that it may cause when mis-used/mis-configured.

This depends on https://github.com/crowbar/crowbar-ha/pull/238

### CODE REVIEW

**Comment by vuntz** [comment]
Note: it's better to view the "big" commit with `diff -w` (or `w=1` in the github url), as it's mostly about changing the indentation.

File: chef/cookbooks/ceilometer/recipes/server_ha.rb [hunk 1]
```diff
@@ -23,57 +23,59 @@
...
 
-  if node[:ceilometer][:use_mongodb]
-    clone_name = "cl-#{primitive_name}"
-    order_name = "o-#{clone_name}-mongo"
-    pacemaker_order order_name do
-      score "Mandatory"
-      ordering "cl-mongodb #{clone_name}"
-      action :update
-      only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
+    if node[:ceilometer][:use_mongodb]
```
[the remark this answers is not in the document]
Comment (vuntz) [reply chef/cookbooks/ceilometer/recipes/server_ha.rb]: For the record, I disagree for this case.

**Comment by vuntz** [comment]
> The code changes look good, but it would be nice if the main commit message also mentioned the things I mentioned before, e.g.

@aspiers so I tried to do this. But probably failed in interesting ways :-) Tell me!

**Comment by guangyee** [comment]
@vuntz I am trying to test the changes. First thing I noticed was the none of the OpenStack service systemd unit files has "Restart" and "RestartSec"' options specified. When I purposely killed a service (i.e. heat-api) vial kill -9, I was under the impression that it will be brought back. But I still see this in the system logs.

Sep 19 22:19:26 d52-54-77-77-01-01 systemd[1]: Stopped OpenStack Heat: native API.
Sep 19 22:19:28 d52-54-77-77-01-01 crmd[20858]:   notice: Result of stop operation for heat-api on d52-54-77-77-01-01: 0 (ok)
Sep 19 22:19:28 d52-54-77-77-01-01 crmd[20858]:   notice: Initiating start operation heat-api_start_0 locally on d52-54-77-77-01-01
Sep 19 22:19:28 d52-54-77-77-01-01 lrmd[20855]:   notice: executing - rsc:heat-api action:start call_id:829
Sep 19 22:19:28 d52-54-77-77-01-01 systemd[1]: Reloading.
Sep 19 22:19:28 d52-54-77-77-01-01 systemd[1]: nss-lookup.target: Dependency Before=nss-lookup.target dropped
Sep 19 22:19:28 d52-54-77-77-01-01 systemd[1]: Started Cluster Controlled openstack-heat-api.
Sep 19 22:19:30 d52-54-77-77-01-01 crmd[20858]:   notice: Result of start operation for heat-api on d52-54-77-77-01-01: 0 (ok)
Sep 19 22:19:30 d52-54-77-77-01-01 crmd[20858]:   notice: Initiating monitor operation heat-api_monitor_10000 locally on d52-54-77-77-01-01
Sep 19 22:19:30 d52-54-77-77-01-01 crmd[20858]:   notice: Transition 117 (Complete=8, Pending=0, Fired=0, Skipped=0, Incomplete=0, Source=/var/lib/pacemaker/pengine/pe-input-104.bz2): Complete
Sep 19 22:19:30 d52-54-77-77-01-01 crmd[20858]:   notice: State transition S_TRANSITION_ENGINE -> S_IDLE

It appears it is still being managed by Pacemaker? Is that still the intention or am I misunderstood the purpose of this PR?

**Comment by vuntz** [comment]
> Hmph, I spelled them out in a bulleted list and you still didn't include them :-) Specifically the 2nd and 4th items. But I can only imagine that this was deliberate for some reason. I can live with the omission of the 4th since hopefully we're not going to forget how to do the improvements in the future, and we're not currently using OCF RAs for decent monitoring anyway. I am slightly worried about the 2nd point, although presumably if the services had major issues with restarting independently of their dependencies then we would have already seen that every time restarts are done via Pacemaker maintenance mode.

Right, the 4th one was deliberate. The 2nd one is kind of part of "we will consider changing the default once we implement Restart= in systemd services if it is found out that this has no negative impact"; not exactly what you said, but I actually wanted to be more generic here. Let me be the annoying person who goes ahead because he's hopeful that it'll be okay ;-)

**Pull request merged**